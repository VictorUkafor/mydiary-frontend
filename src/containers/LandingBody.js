import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import man from '../assets/man1.png';
import landingLogo from '../assets/landing-logo.png';


export class LandingBody extends Component {
  linking(link, message) {
    return (
      <Link to={`/${link}`}>
        <button className="direct" type="button">{message}</button>
      </Link>
    );
  }

  landingButton(authLink, authMessage, defaultLink, defaultMessage) {
    const { user } = this.props;
    if (user) {
      this.linking(authLink, authMessage);
    }
    this.linking(defaultLink, defaultMessage);
  }

  render() {
    return (
      <div className="body">
        <div className="landing">
          <div className="image-ad">
            <img src={man} alt="" />
          </div>
          <div className="add">
            <img src={landingLogo} alt="" />
            <h1 className="add-h1">You&apos;ve got any idea, thought or plan!
              <br />
          Pen it down on <span className="x-h1">MyDiary</span> now
            </h1>
            <div className="d-buttons">
              {this.landingButton('add-entry', 'Add an entry', 'login', 'Have an account? Sign In')}
              {this.landingButton('entries', 'View all entries', 'signup', 'You don\'t? Sign up today!')}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  user: state.user,
});

export default connect(mapStateToProps)(LandingBody);
