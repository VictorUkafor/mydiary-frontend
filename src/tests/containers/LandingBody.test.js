import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LandingBody from '../../containers/LandingBody';

test('<LandingBody />', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<LandingBody />);
  const result = renderer.getRenderOutput();

  expect(result.firstChild).toMatchSnapshot();
});
