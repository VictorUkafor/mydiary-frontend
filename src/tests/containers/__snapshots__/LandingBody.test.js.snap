// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingBody /> 1`] = `undefined`;
